### DISCUSSION TITLE
umi升级到umi4后，ProForm里面引入了自定义组件<Form.Item>，拿不到自定义组件的字段值

### DISCUSSION THREAD

**Discussion by titld** [umijs/umi#10001] [Q&A]
在proform里面引入了自定义组件<Form.Item>， 但是在onValueChange方法里面拿不到这个参数，应该怎么解决呢？在umi3中可以的，现在换成了umi4版本就拿不到了；我期望onValueChange方法能够拿到所有的字段（包括自定义组件的switch字段）;
© 版本信息
ProComponents 版本: 2.3.19
ProForm 版本：2.2.7
umi 版本 4.0.34
浏览器环境
开发环境 [e.g. mac OS]
🚑 其他信息
```js
<ProForm
 onValuesChange={(changeValues) => console.log(changeValues)}
>
  <ProFormText name="name" label="name" />
 <Form.Item name="switch" label="Switch" valuePropName="checked">
    <Switch />
  </Form.Item>
</ProForm>
```

**Comment by fz6m** [comment]
pro component 不是啥都有吗，不用自己定义：https://procomponents.ant.design/components/field-set

**Reply by titld** [reply]
原子组件不满足项目要求，需要单独封装，之前的umi3版本没问题，升级到umi4之后就不行了；

**Reply by fz6m** [reply]
换成 `ProForm.Item`

**Reply by titld** [reply]
不行 效果一样

**Reply by fz6m** [reply]
到 [pro-components](https://github.com/ant-design/pro-components) 反馈一下你的诉求吧，具体的自定义方法我不明白。

**Reply by titld** [reply]
我表述有问题，新开了一个issue#10019，有空帮忙再看看呢
